Patch origin: in-house
Patch status: Solaris-specific; not suitable for upstream

Download/use only archives we really need.
Don't install unnecessary stuff.

SAAJ API should not be needed as we still build with JDK 8
(and not with JDK 11 or above).

diff --git a/build.xml b/build.xml
index 5db4081..10c97bd 100644
--- a/build.xml
+++ b/build.xml
@@ -219,7 +219,6 @@
   <!-- Classpaths -->
   <path id="compile.classpath">
     <pathelement location="${jdt.jar}"/>
-    <pathelement location="${saaj-api.jar}"/>
   </path>
 
   <path id="tomcat.classpath">
@@ -1078,8 +1077,6 @@
   <target name="deploy" depends="package,build-docs,build-tomcat-jdbc,compile-webapp-examples"
           description="Default. Builds a working Tomcat instance">
 
-    <copy tofile="${tomcat.build}/bin/commons-daemon.jar" file="${commons-daemon.jar}" />
-
     <!-- Copy scripts -->
     <copy todir="${tomcat.build}/bin">
       <fileset dir="bin">
@@ -1851,12 +1848,6 @@
       </fileset>
     </copy>
 
-    <!-- Source bundles for native components -->
-    <copy tofile="${tomcat.dist}/bin/tomcat-native.tar.gz"
-            file="${tomcat-native.tar.gz}" />
-    <copy tofile="${tomcat.dist}/bin/commons-daemon-native.tar.gz"
-            file="${commons-daemon.native.src.tgz}" />
-
     <!-- platform README files -->
     <echo append="false" file="${tomcat.dist}/bin/x64/README">
 Apache Tomcat ${version} native binaries for Win64 AMD64/EMT64 platform.
@@ -2753,16 +2744,6 @@ skip.installer property in build.properties" />
   <target name="download-compile"
           description="Download components necessary to compile" >
 
-    <!-- Download Commons Daemon -->
-    <antcall target="downloadgz-2">
-      <param name="sourcefile.1" value="${commons-daemon.bin.loc.1}"/>
-      <param name="sourcefile.2" value="${commons-daemon.bin.loc.2}"/>
-      <param name="destfile" value="${commons-daemon.jar}"/>
-      <param name="checksum.enabled" value="${commons-daemon.bin.checksum.enabled}"/>
-      <param name="checksum.algorithm" value="${commons-daemon.bin.checksum.algorithm}"/>
-      <param name="checksum.value" value="${commons-daemon.bin.checksum.value}"/>
-    </antcall>
-
     <!-- Download JDT (Eclipse compiler) -->
     <antcall target="downloadfile-2">
       <param name="sourcefile.1" value="${jdt.loc.1}"/>
@@ -2774,16 +2755,6 @@ skip.installer property in build.properties" />
       <param name="checksum.value" value="${jdt.checksum.value}"/>
     </antcall>
 
-    <!-- Download SAAJ API -->
-    <antcall target="downloadfile">
-      <param name="sourcefile" value="${saaj-api.loc}"/>
-      <param name="destfile" value="${saaj-api.jar}"/>
-      <param name="destdir" value="${saaj-api.home}"/>
-      <param name="checksum.enabled" value="${saaj-api.checksum.enabled}"/>
-      <param name="checksum.algorithm" value="${saaj-api.checksum.algorithm}"/>
-      <param name="checksum.value" value="${saaj-api.checksum.value}"/>
-    </antcall>
-
   </target>
 
   <target name="download-test-compile"
@@ -2867,16 +2838,6 @@ skip.installer property in build.properties" />
   <target name="download-dist"
           description="Download additional components for a distribution" >
 
-    <antcall target="downloadfile-2">
-      <param name="sourcefile.1" value="${tomcat-native.loc.1}"/>
-      <param name="sourcefile.2" value="${tomcat-native.loc.2}"/>
-      <param name="destfile" value="${tomcat-native.tar.gz}"/>
-      <param name="destdir" value="${tomcat-native.home}"/>
-      <param name="checksum.enabled" value="${tomcat-native.src.checksum.enabled}"/>
-      <param name="checksum.algorithm" value="${tomcat-native.src.checksum.algorithm}"/>
-      <param name="checksum.value" value="${tomcat-native.src.checksum.value}"/>
-    </antcall>
-
     <antcall target="downloadzip-2">
       <param name="sourcefile.1" value="${tomcat-native.win.1}"/>
       <param name="sourcefile.2" value="${tomcat-native.win.2}"/>
@@ -2887,16 +2848,6 @@ skip.installer property in build.properties" />
       <param name="checksum.value" value="${tomcat-native.win.checksum.value}"/>
     </antcall>
 
-    <antcall target="downloadfile-2">
-      <param name="sourcefile.1" value="${commons-daemon.native.src.loc.1}"/>
-      <param name="sourcefile.2" value="${commons-daemon.native.src.loc.2}"/>
-      <param name="destfile" value="${commons-daemon.native.src.tgz}"/>
-      <param name="destdir" value="${commons-daemon.home}"/>
-      <param name="checksum.enabled" value="${commons-daemon.native.src.checksum.enabled}"/>
-      <param name="checksum.algorithm" value="${commons-daemon.native.src.checksum.algorithm}"/>
-      <param name="checksum.value" value="${commons-daemon.native.src.checksum.value}"/>
-    </antcall>
-
     <antcall target="downloadzip-2">
       <param name="sourcefile.1" value="${commons-daemon.native.win.loc.1}"/>
       <param name="sourcefile.2" value="${commons-daemon.native.win.loc.2}"/>
